docs/user/datastores/query_interceptor.rst

```diff
@@ -49,7 +49,7 @@ Interceptors must have a default, no-arg constructor. The interceptor lifecycle
 
 1. The instance is instantiated via reflection, using its default constructor
 #. The instance is initialized via the ``init`` method, passing in the data store containing the simple feature type
-#. ``rewrite`` and ``guard`` are called once for each query
+#. ``rewrite`` and ``guard`` are called once for each query (these two methods must be thread-safe)
```

Comment:
Did you hit a non-thread safe case as you thinking through things?

Reply:
just clarifying that if multiple queries come in at once that method might be invoked from 2 different threads at the same time. i think all our impls were fine.